opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/filter/FilterQueryBuilder.java

```diff
@@ -57,7 +58,7 @@ public class FilterQueryBuilder extends ExpressionNodeVisitor<QueryBuilder, Obje
           .put(BuiltinFunctionName.GREATER.getName(), new RangeQuery(Comparison.GT))
           .put(BuiltinFunctionName.LTE.getName(), new RangeQuery(Comparison.LTE))
           .put(BuiltinFunctionName.GTE.getName(), new RangeQuery(Comparison.GTE))
-          .put(BuiltinFunctionName.LIKE.getName(), new WildcardQuery())
+          .put(BuiltinFunctionName.LIKE.getName(), new LikeQuery())
```

Comment:
Why not use WildcardQuery?

Reply:
Currently `LIKE` uses WildcardQuery.java in V2 while `wildcard_query` uses WildcardQuery.java in legacy. We want to classify `wildcard_query` as a relevance function however, since LIKE in V2 already uses WildcardQuery.java then it had to be separated as it uses `ReferenceExpression` instead of `NamedArgumentExpression` which is used in relevance functions. Making `LIKE` use `wildcard_query` results in a `ClassCastException`.

`LIKE` also converts text fields to keyword while `wildcard_query` in the legacy does not. Thus, `LIKE` using WildcardQuery as a relevance function would be a breaking change. If we want `wildcard_query` to convert text fields to keyword, then we would want to make it so for all relevance functions and can be a separate issue to enhance relevance functions.

Reply:
1. In SQL engine, like is simplified version of wildcard_query, right? If so, like and wildcard_query should rewrite as OpenSearch wildcard_query. The differences are, (1) rewrite Like to wildcard query and use default parameter (2) rewrite wildcard_query to wildcard query based on user provided parameters.
2. I think breaking change maybe make sense. Only field of keyword type could use LIKE operator. If it is multitype field, user need to explict specifiy it.

Reply:
This breaking change is tracked with https://github.com/opensearch-project/sql/issues/1143 and should be implemented in `3.x`. This can be merged in `2.x` to allow for legacy deprecation.